tutorials-v5/miscellaneous/v5_paper-jax.md

```diff
@@ -0,0 +1,327 @@
+GPUs are good at parallelizing many small matrix-vector operations, such as integrating small systems across multiple parameters or simulating quantum circuits with repeated small matrix operations.
+For a single ODE involving large matrices, the advantages are less straightforward since ODE solvers are inherently sequential.
+However, as it is illustrated in the [QuTiPv5 paper](#References), there is a cross-over point at which using JAX is beneficial.
+
+### 1D Ising Spin Chain
+
+To illustrate the usage of QuTiP-JAX, we look at the one-dimensional spin chain with the Hamiltonian
+
+$H = \sum_{i=1}^N g_0 \sigma_z^{(n)} - \sum_{n=1}^{N-1} J_0 \sigma_x^{(n)} \sigma_x^{(n+1)}$.
+
+We hereby consider $N$ spins that share an energy splitting of $g_0$ and have a coupling strength $J_0$.
+The end of the chain connects to an environment described by a Lindbladian dissipator which we model with the collapse operator $\sigma_x^{(N-1)}$ and coupling rate $\gamma$.
+
+As part of the [QuTiPv5 paper](#References), we see an extensive study on the computation time depending on the dimensionality $N$.
+In this example we cannot replicate the performance of a supercomputer of course, so we rather focus on the correct implementation to solve the Lindblad equation for this system using JAX and `mesolve()`.
+
+```python
+# system parameters
+N = 4  # number of spins
+g0 = 1  # energy splitting
+J0 = 1.4  # coupling strength
+gamma = 0.1  # dissipation rate
+
+# simulation parameters
+tlist = jnp.linspace(0, 5, 100)
+opt = {
+    "normalize_output": False,
+    "store_states": True,
+    "method": "diffrax",
+    "stepsize_controller": PIDController(
+        rtol=settings.core["rtol"], atol=settings.core["atol"]
+    ),
+    "solver": Tsit5(),
+}
+```
+
+```python
+with CoreOptions(default_dtype="CSR"):
+    # Operators for individual qubits
+    sx_list, sy_list, sz_list = [], [], []
+    for i in range(N):
+        op_list = [qeye(2)] * N
+        op_list[i] = sigmax()
+        sx_list.append(tensor(op_list))
+        op_list[i] = sigmay()
+        sy_list.append(tensor(op_list))
+        op_list[i] = sigmaz()
+        sz_list.append(tensor(op_list))
+
+    # Hamiltonian - Energy splitting terms
+    H = 0.0
+    for i in range(N):
+        H += g0 * sz_list[i]
+
+    # Interaction terms
+    for n in range(N - 1):
+        H += -J0 * sx_list[n] * sx_list[n + 1]
+
+    # Collapse operator acting locally on single spin
+    c_ops = [gamma * sx_list[N - 1]]
+
+    # Initial state
+    state_list = [basis(2, 1)] * (N - 1)
+    state_list.append(basis(2, 0))
+    psi0 = tensor(state_list)
+
+    result = mesolve(H, psi0, tlist, c_ops, e_ops=sz_list, options=opt)
+```
+
+```python
+for i, s in enumerate(result.expect):
+    plt.plot(tlist, s, label=rf"$n = {i+1}$")
+plt.xlabel("Time")
+plt.ylabel(r"$\langle \sigma^{(n)}_z \rangle$")
+plt.legend()
+plt.show()
+```
+
```

Comment:
Could add some funky animation plot!

```
%matplotlib notebook
from IPython.display import HTML
ax, ani = qt.anim_matrix_histogram(result_ising)
HTML(ani.to_jshtml(fps=15))
```

Reply:
ah ouch.  i tried a different example the other day and it was fairly slow too. we could reduce the number of points in tlist, but perhaps just better not to add here!

Reply:
From my experience, a long `tlist` usually causes a long compilation time, because we have a for loop in the qutip base solver. This for loop will be flattened by jax compilation. This can be speeded up by rewriting it with a JAX loop. But I don't know how hard it is because in each step we add something to the result class.